homeassistant/components/device_tracker/checkin.py

```diff
@@ -0,0 +1,110 @@
+"""
+homeassistant.components.device_tracker.checkin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Checkin platform for the device tracker.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.checkin.html
+"""
+
+import logging
+
+from homeassistant.const import (
+    HTTP_UNPROCESSABLE_ENTITY, HTTP_INTERNAL_SERVER_ERROR)
+from homeassistant.const import (ATTR_LATITUDE, ATTR_LONGITUDE)
+    
+DEPENDENCIES = ['http', 'zone']
+
+_SEE = 0
+_HASS = None
+
+URL_API_CHECKIN_ENDPOINT = "/api/checkin"
+
+
+def setup_scanner(hass, config, see):
+    """ Set up an endpoint for the checkin app. """
+
+    # Use a global variable to keep setup_scanner compact when using a callback
+    global _SEE
+    _SEE = see
+    
+    global _HASS
+    _HASS = hass
+
+    # POST would be semantically better, but that currently does not work
+    # since Geofancy sends the data as key1=value1&key2=value2
+    # in the request body, while Home Assistant expects json there.
+
+    hass.http.register_path(
+        'GET', URL_API_CHECKIN_ENDPOINT, _handle_get_api_checkin)
+
+    return True
+
+
+def available_zones():
```

Comment:
Is the zone platform enabled even if no zones are defined? Could it happen that if someone didn't define any zones, the dependency is not met and this component won't be loaded?

If that can happen, would it be better to remove the zone dependency?

If it cannot, I'll do what you said.